### ISSUE TITLE
Show connector FailureReason when setting up a connection

### ISSUE THREAD

**Issue by pedroslopez** [airbytehq/airbyte#16840]
## Tell us about the problem you're trying to solve

When the user is setting up their source, if the connector fails when calling `discover`, the error will not be shown to the user. This means that they can't debug the problem.

This is what the user sees right now:

<img width="1101" alt="image" src="https://user-images.githubusercontent.com/4368928/190830411-5379a89b-c691-485a-a29b-7deb0f5520f9.png">

It also means that the only way we can see the actual error is by looking at the created issue in Sentry.

## Describe the solution you’d like

Since we're already building up the FailureReason for these synchronous commands, we can propagate that to the api response and display it accordingly in the UI.

This is also the case for schema refreshes, so may be related to https://github.com/airbytehq/airbyte/issues/14096 - not sure how different these codepaths are

**Comment by krishnaglick** [comment]
There may be some logic overlap between this and #14096, but it'd likely be around extracting the error. Surfacing that to the user should be a bit different, but not overly complicated I imagine.

**Comment by evantahler** [comment]
@pedroslopez is this a duplicate of https://github.com/airbytehq/airbyte/issues/18514 perhaps?

**Comment by pedroslopez** [comment]
@evantahler err, sort of.. when I wrote this initially I did include the API piece that #18514 is about, but this is also about showing that in the frontend when running the initial discover call. https://github.com/airbytehq/airbyte/issues/18407 is the UI equivalent for the CHECK call, but this one's for discover. We should also make sure it shows up when running discover both in the new connector set up flow and in the schema refresh flow.
